Source/RapyutaSimulationPlugins/Private/Core/RRUObjectUtils.cpp

```diff
@@ -145,6 +149,65 @@ ARRBaseActor* URRUObjectUtils::SpawnSimActor(UWorld* InWorld,
     return newActor;
 }
 
+void URRUObjectUtils::GetActorCenterAndBoundingBoxVertices(const AActor* InActor,
+                                                           const AActor* InBaseActor,
+                                                           TArray<FVector>& OutCenterAndVertices,
+                                                           bool bInIncludeNonColliding)
+{
+    ARRGameState* gameState = URRCoreUtils::GetGameState<ARRGameState>(InActor);
+
+    // [InActor]'s Local bounding box
+    FVector centerLocal, extentsLocal;
+    FBox actorLocalBox = InActor->CalculateComponentsBoundingBoxInLocalSpace(bInIncludeNonColliding);
+    actorLocalBox.GetCenterAndExtents(centerLocal, extentsLocal);
+
+    // [InActor]'s global bounding box with rotation (in BaseActor frame or World frame)
+    // Reference : DrawDebugBox()
+    // (snote) The exact order of points is important. Consult with Robot Research team before adjusting.
+
+    // Transfom [centerLocal] to global
+    // UnrealEngine/Engine/Plugins/2D/Paper2D/Source/Paper2D/Private/PaperSpriteComponent.cpp:179
+    const FTransform baseTransform =
+        InBaseActor ? InActor->GetActorTransform().GetRelativeTransform(InBaseActor->GetTransform()) : InActor->GetTransform();
+    const FVector center = baseTransform.TransformPosition(centerLocal);
+    OutCenterAndVertices.Emplace(center);
+
+    // [InActor]'s global Rotation (in BaseActor frame or World frame)
+    const FQuat& actorQuat = InBaseActor ? URRUObjectUtils::GetRelativeQuatFrom(InActor, InBaseActor) : InActor->GetActorQuat();
+
+    // Calculate global vertices as : [center] + Rotated [ExtentsLocal]
+    for (auto i = 0; i < 8; ++i)
+    {
+        const FVector& vertexNormal = gameState->ENTITY_BOUNDING_BOX_VERTEX_NORMALS[i];
+        OutCenterAndVertices.Emplace(center +
+                                     actorQuat.RotateVector(FVector((0.f == vertexNormal.X) ? -extentsLocal.X : extentsLocal.X,
+                                                                    (0.f == vertexNormal.Y) ? -extentsLocal.Y : extentsLocal.Y,
+                                                                    (0.f == vertexNormal.Z) ? -extentsLocal.Z : extentsLocal.Z)));
+    }
+#if RAPYUTA_SIM_VISUAL_DEBUG
+    if (IsInGameThread() && (nullptr == InBaseActor))
+    {
+        // Draw [InActor]'s bounding box before projecting it onto screen to get the 2D coordinates.
+        OutCenterAndVertices.RemoveAt(0);
```

Comment:
Why remove this then insert at the same index rather than just replacing the value at index 0? This will do at lest two memmoves of all remaining vertices in that array.